lib/spack/spack/build_systems/python.py

```diff
@@ -359,30 +365,11 @@ def check_args(self, spec, prefix):
 
     # Testing
 
-    def build_test(self):
-        """Run unit tests after in-place build.
-
-        These tests are only run if the package actually has a 'test' command.
-        """
-        if self._setup_command_available('test'):
-            args = self.test_args(self.spec, self.prefix)
-
-            self.setup_py('test', *args)
```

Comment:
Running `python setup.py test` is deprecated, as most packages tell you to use something like `tox` instead. Also, the vast majority of packages do not upload any test files to PyPI, making these tests a no-op. Even if they do, I rarely manage to get the tests to pass, and many tests introduce circular dependencies. I've been meaning to do this for a long time now, but I'm removing these tests for the aforementioned reasons.